runtime/gc_base/StringTable.cpp

```diff
@@ -642,48 +642,37 @@ j9gc_createJavaLangString(J9VMThread *vmThread, U_8 *data, UDATA length, UDATA s
 
 		J9VMJAVALANGSTRING_SET_VALUE(vmThread, result, charArray);
 
-		if (IS_STRING_COMPRESSION_ENABLED_VM(vm)) {
-			if (isCompressable) {
-				if (J2SE_VERSION(vm) >= J2SE_V10) {
-					J9VMJAVALANGSTRING_SET_CODER(vmThread, result, 0);
-				}
-
-				J9VMJAVALANGSTRING_SET_COUNT(vmThread, result, (I_32) unicodeLength);
+		if (IS_STRING_COMPRESSION_ENABLED_VM(vm) && isCompressable) {
+			if (J2SE_VERSION(vm) >= J2SE_19) {
+				J9VMJAVALANGSTRING_SET_CODER(vmThread, result, 0);
 			} else {
-				if (J9VMJAVALANGSTRING_COMPRESSIONFLAG(vmThread, stringClass) == 0) {
-					/*
-					 * jitHookClassPreinitialize will process initialization events for String compression sideEffectGuards
-					 * so we must initialize the class if this is the first time we are loading it
-					 */
-			 		J9Class* flagClass = vm->internalVMFunctions->internalFindKnownClass(vmThread, J9VMCONSTANTPOOL_JAVALANGSTRINGSTRINGCOMPRESSIONFLAG, J9_FINDKNOWNCLASS_FLAG_INITIALIZE);
-
-					if (flagClass == NULL) {
-						goto nomem;
-					} else {
-						j9object_t flag = J9AllocateObject(vmThread, flagClass, allocateFlags);
+				J9VMJAVALANGSTRING_SET_COUNT(vmThread, result, (I_32)unicodeLength);
+			}
+		} else {
+			if (J2SE_VERSION(vm) >= J2SE_19) {
+				J9VMJAVALANGSTRING_SET_CODER(vmThread, result, 1);
+			} else {
+				J9VMJAVALANGSTRING_SET_COUNT(vmThread, result, (I_32)unicodeLength);
```

Comment:
The original code had a case which set the high bit for uncompressed strings.

Reply:
Refactored and fixed in b506e23